### PR TITLE
React lazy loading

### PR DESCRIPTION
**Pull request by debbieellis7**
## Description
Webpack, Babel, and React setup for lazy loading.
Issue: #889
This modification is part of issue #889 

## Checklist:
- [x] I've tested the code.
- [x] My code is easy to read, follow, and understand
- [x] My code follows the accessibility standards. 
- [x] My code has proper inline documentation / docblocks.

### CODE REVIEW

File: webpack.config.js [hunk 1]
```diff
@@ -2,6 +2,8 @@ const webpack = require('webpack')
 const path = require('path')
 const UglifyJSPlugin = require('uglifyjs-webpack-plugin')
 const PROD = (process.env.NODE_ENV === 'production')
+const pluginDirectory = path.basename(__dirname)
+const chunkPath = `/wp-content/plugins/${pluginDirectory}/dist/assets/js/`
```
Comment (jakejackson1) [inline webpack.config.js]: We need to be able to get rid of `/wp-content/plugins/` in the config.

Comment (debbieellis7) [reply webpack.config.js]: I think that's the simplest way possible. Here is another way.

const pluginDirectory = path.basename(__dirname)
const wpPath = path.basename(path.resolve('../../'))
const plgPath = path.basename(path.resolve('../'))
const publicPath = `/${wpPath}/${plgPath}/${pluginDirectory}/dist/assets/js/`

**Comment by jakejackson1** [comment]
@debbieellis7 I've made some modifications to this PR to allow the public path to be defined at runtime. This resolves the issue I brought up in my previous review. 

Could you run through it, fix up any linting issues, verify it works as expected, and squash ect?

**Comment by debbieellis7** [comment]
> @debbieellis7 I've made some modifications to this PR to allow the public path to be defined at runtime. This resolves the issue I brought up in my previous review.
> 
> Could you run through it, fix up any linting issues, verify it works as expected, and squash ect?

All good!

**Pull request merged**